### PR TITLE
fix libiosexec for tvOS

### PR DESCRIPTION
**Pull request by asdfugil**

### CODE REVIEW

**Comment by CRKatri** [comment]
Could you add watchos and bridgeos (if possible) too?

**Comment by asdfugil** [comment]
simulator target would make more sense then bridgeos
there's a bridgeos target for tbd so sure but what is the compiler flag to compile for bridgeOS
also where's bridgeOS.sdk

**Comment by CRKatri** [comment]
Hence why I said it's ridiculous

**Comment by asdfugil** [comment]
```
clang: warning: using sysroot for 'MacOSX' but targeting 'Bridge' [-Wincompatible-sysroot]
```
uhh

**Comment by asdfugil** [comment]
I will add watchOS target here

bridgeOS will be addressed in another upcoming PR that allows compiling for bridgeos-arm64

**Comment by Diatrus** [comment]
Support watchOS here and I'll merge it.

**Comment by CRKatri** [comment]
This isn't working for me
```
➜ aarch64-apple-darwin-clang -Os -g -arch arm64_32 -isysroot $WATCHOS_SYSROOT test.c -v libiosexec.tbd   
clang version 12.0.0
Target: aarch64_32-apple-darwin
Thread model: posix
InstalledDir: /usr/local/bin
 "/usr/local/bin/aarch64-apple-darwin-clang" -cc1 -triple aarch64_32-apple-watchos7.2.0 -Wundef-prefix=TARGET_OS_ -Werror=undef-prefix -Wdeprecated-objc-isa-usage -Werror=deprecated-objc-isa-usage -Werror=implicit-function-declaration -emit-obj --mrelax-relocations -disable-free -disable-llvm-verifier -discard-value-names -main-file-name test.c -mrelocation-model pic -pic-level 2 -mframe-pointer=non-leaf -fno-rounding-math -munwind-tables -target-sdk-version=7.2 -fcompatibility-qualified-id-block-type-checking -fvisibility-inlines-hidden-static-local-var -target-cpu apple-s4 -target-feature +v8.3a -target-feature +fp-armv8 -target-feature +neon -target-feature +crc -target-feature +crypto -target-feature +fullfp16 -target-feature +ras -target-feature +lse -target-feature +rdm -target-feature +rcpc -target-feature +zcm -target-feature +zcz -target-feature +sha2 -target-feature +aes -target-abi darwinpcs -fallow-half-arguments-and-returns -debug-info-kind=standalone -dwarf-version=4 -debugger-tuning=lldb -v -resource-dir /usr/local/lib/clang/12.0.0 -isysroot /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk -internal-isystem /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/local/include -internal-isystem /usr/local/lib/clang/12.0.0/include -internal-externc-isystem /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/include -Os -fdebug-compilation-dir /usr/home/cameron/Documents/iOS/libiosexec -ferror-limit 19 -stack-protector 1 -fblocks -fencode-extended-block-signature -fregister-global-dtors-with-atexit -fgnuc-version=4.2.1 -fmax-type-align=16 -fcolor-diagnostics -vectorize-loops -vectorize-slp -o /tmp/test-699f6a.o -x c test.c
clang -cc1 version 12.0.0 based upon LLVM 12.0.0 default target x86_64-portbld-freebsd14.0
ignoring nonexistent directory "/usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/local/include"
ignoring nonexistent directory "/usr/local/lib/clang/12.0.0/include"
ignoring nonexistent directory "/usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/Library/Frameworks"
#include "..." search starts here:
#include <...> search starts here:
 /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/include
 /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/System/Library/Frameworks (framework directory)
End of search list.
 "/usr/local/bin/aarch64-apple-darwin-ld" -dynamic -arch arm64_32 -watchos_version_min 7.2.0 -syslibroot /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk -o a.out /tmp/test-699f6a.o libiosexec.tbd -lSystem
ld: warning: ignoring file libiosexec.tbd, missing required architecture unknown in file libiosexec.tbd (2 slices)
ld: warning: ignoring file /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/lib/libSystem.tbd, missing required architecture unknown in file /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/lib/libSystem.tbd
Undefined symbols for architecture arm64_32:
  "_ie_execl", referenced from:
      _main in test-699f6a.o
ld: symbol(s) not found for architecture arm64_32
aarch64-apple-darwin-clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by asdfugil** [comment]
Do you fr think libSystem.tbd is broken?
```
ld: warning: ignoring file libiosexec.tbd, missing required architecture unknown in file libiosexec.tbd (2 slices)
ld: warning: ignoring file /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/lib/libSystem.tbd, missing required architecture unknown in file /usr/home/cameron/Documents/iOS/SDK/WatchOS7.2.sdk/usr/lib/libSystem.tbd
```
> architecture unknown

Looks more like your problem

**Comment by asdfugil** [comment]
Also works on my machine.

**Comment by asdfugil** [comment]
anyways I can investigate this a little bit

**Comment by CRKatri** [comment]
It may just be slightly old cctools/tapi acting up

File: build_misc/libiosexec/libiosexec.1.tbd [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+  - target:          arm64-bridgeos
+    value:           AC079058-D4B0-11EB-A862-63D08897B634
+
 flags:           [ not_app_extension_safe ]
 install-name:    '/usr/lib/libiosexec.1.dylib'
 current-version: 0
 compatibility-version: 0
 exports:
-  - targets:         [ arm64-ios ]
+  - targets:         [ arm64-ios, arm64-tvos ]
```
Comment (CRKatri) [inline build_misc/libiosexec/libiosexec.1.tbd:23]: Is this missing targets or is that intentional?

**Review by CRKatri — APPROVED** [review]

**Pull request merged**